src/os/bluestore/BlueStore.cc

```diff
@@ -6445,7 +6474,9 @@ void BlueStore::_txc_state_proc(TransContext *txc)
       }
       txc->shared_blobs_written.clear();
       if (cct->_conf->bluestore_sync_submit_transaction &&
-	  fm->supports_parallel_transactions()) {
+	  fm->supports_parallel_transactions() &&
+          (1 == kv_sync_thread_list.size())) {
```

Comment:
I think this idea is similar as what OSD ShardWQthread  did and one pg IOs will go one shard, use kv_lock_shard guarantee pg consistency.  so needn't all IOs goes to one kv_sync_thread. we only need guarantee sequence in one PG.